src/drivers/cmakeLegacyDriver.ts

```diff
@@ -85,14 +85,23 @@ export class CMakeLegacyDriver extends CMakeDriver {
         const args = Array.from(args_);
         args.push(util.lightNormalizePath(this.sourceDir));
 
-        const gen = this.generator;
-        if (gen) {
-            args.push(`-G${gen.name}`);
-            if (gen.toolset) {
-                args.push(`-T${gen.toolset}`);
+        if (configurePreset && configurePreset.generator) {
+            args.push('-G');
```

Comment:
Same feedback as the file api driver. Construct `const gen` from the preset and use it here instead of having two copies of the code.

Reply:
Thank you, the legacyDriver needed some other changes too.